src/test/test_ekubo_oracle.cairo

```diff
@@ -0,0 +1,181 @@
+        singleton::{ISingletonDispatcherTrait, ISingletonDispatcher},
+        test::{
+            setup::{create_pool_v3, deploy_asset_with_decimals, setup_env_v3, TestConfigV3, EnvV3},
+            mock_ekubo_oracle::{IMockEkuboOracleDispatcher, IMockEkuboOracleDispatcherTrait}
+        },
+        extension::{
+            interface::{IExtensionDispatcher, IExtensionDispatcherTrait},
+            default_extension_ek::{IDefaultExtensionEKDispatcherTrait, IDefaultExtensionEKDispatcher,}
+        },
+        math::pow_10, vendor::erc20::{ERC20ABIDispatcher as IERC20Dispatcher, ERC20ABIDispatcherTrait},
+    };
+
+    fn setup_env_v3_with_asset_decimals(
+        collateral_asset_decimals: u32, debt_asset_decimals: u32, third_asset_decimals: u32, quote_asset_decimals: u32
+    ) -> EnvV3 {
+        let class = declare("MockAsset");
+        let lender = contract_address_const::<'lender'>();
+
+        let decimals = collateral_asset_decimals;
+        let supply = 100 * pow_10(decimals);
+        let calldata = array![
+            'Collateral', 'COLL', collateral_asset_decimals.into(), supply.low.into(), supply.high.into(), lender.into()
+        ];
+        let collateral_asset = IERC20Dispatcher { contract_address: class.deploy(@calldata).unwrap() };
+
+        let decimals = debt_asset_decimals;
+        let supply = 100 * pow_10(decimals);
+        let calldata = array![
+            'Debt', 'DEBT', debt_asset_decimals.into(), supply.low.into(), supply.high.into(), lender.into()
+        ];
+        let debt_asset = IERC20Dispatcher { contract_address: class.deploy(@calldata).unwrap() };
+
+        let decimals = third_asset_decimals;
+        let supply = 100 * pow_10(decimals);
+        let calldata = array![
+            'Third', 'THIRD', third_asset_decimals.into(), supply.low.into(), supply.high.into(), lender.into()
+        ];
+        let third_asset = IERC20Dispatcher { contract_address: class.deploy(@calldata).unwrap() };
+
+        let decimals = quote_asset_decimals;
+        let supply = 100 * pow_10(decimals);
+        let calldata = array![
+            'Quote', 'QUOTE', quote_asset_decimals.into(), supply.low.into(), supply.high.into(), lender.into()
+        ];
+        let quote_asset = IERC20Dispatcher { contract_address: class.deploy(@calldata).unwrap() };
+
+        setup_env_v3(
+            Zeroable::zero(),
+            collateral_asset.contract_address,
+            debt_asset.contract_address,
+            third_asset.contract_address,
+            quote_asset.contract_address,
+            Zeroable::zero(),
+            Zeroable::zero(),
+        )
+    }
+
+    // Test Ekubo price calculation in the case where quote token has less decimals than assets
+    // with actual values from Ekubo's oracle extension.
+    // - collateral_asset (18 decimals): ETH
+    // - third_asset (8 decimals): WBTC
+    // - quote_asset (6 decimals): USDC
+    #[test]
+    fn test_get_ekubo_price_quote_asset_less_decimals() {
```

Comment:
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExajlxeDk3NmR6bmttcXZyMmpucmtvYmcydHRpcGswdjkyZnhxYnZ6OCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/DjolzwbnW9pGAOatK9/giphy.gif)

Reply:
I had to convince myself the conversion formula is correct :smile: